Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -395,107 +517,127 @@ namespace AzToolsFramework
         }
 
         LocalViewBookmarkComponent* bookmarkComponent = containerEntity->FindComponent<LocalViewBookmarkComponent>();
-        if (bookmarkComponent)
+        if (!bookmarkComponent)
         {
-            return bookmarkComponent;
+            // if we didn't find a component then we add it and return it.
+            containerEntity->Deactivate();
+            bookmarkComponent = containerEntity->CreateComponent<LocalViewBookmarkComponent>();
+            containerEntity->Activate();
         }
 
-        // if we didn't find a component then we add it and return it.
-        containerEntity->Deactivate();
-        bookmarkComponent = containerEntity->CreateComponent<LocalViewBookmarkComponent>();
-        containerEntity->Activate();
-
         AZ_Assert(bookmarkComponent, "Couldn't create LocalViewBookmarkComponent.");
+
+        // record an undo step if a local view bookmark component is added and configured
+        ScopedUndoBatch undoBatch("SetupLocalViewBookmarks");
+        undoBatch.MarkEntityDirty(containerEntityId);
+
+        // if the field is empty, we don't have a file linked to the prefab, so we create one and we save it in the component
+        if (bookmarkComponent->GetLocalBookmarksFileName().empty())
+        {
+            bookmarkComponent->SetLocalBookmarksFileName(GenerateBookmarkFileName());
+        }
+
+        if (const auto localBookmarksFileName = bookmarkComponent->GetLocalBookmarksFileName(); !m_fileExistsFn(localBookmarksFileName))
+        {
+            auto registry = AZ::SettingsRegistry::Get();
+            if (!registry)
+            {
+                AZ_Warning("LocalViewBookmarkLoader", false, "Unable to access global settings registry.");
+                return nullptr;
+            }
+
+            // initialize default locations to zero, this is a temporary solution to match the 12 locations of the legacy system
+            // once there is a UI for the view bookmarks these lines should be removed
+            if (const auto setRegPath = LocalViewBookmarkSetRegPath(localBookmarksFileName);
+                !registry->SetObject(setRegPath, AZStd::vector<ViewBookmark>(DefaultViewBookmarkCount, ViewBookmark())))
+            {
+                AZ_Warning("LocalViewBookmarkLoader", false, "SetObject for \"%s\" failed", setRegPath.c_str());
```

Comment:
Should this return `nullptr` like the previous condition does?

Reply:
Agreed it wouldn't hurt, can update 👍